nuxeo-vision-aws/src/main/java/org/nuxeo/vision/aws/AmazonRekognitionProvider.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * (C) Copyright 2017 Nuxeo SA (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Remi Cattiau
+ *     Michael Vachette
+ */
+package org.nuxeo.vision.aws;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.rekognition.AmazonRekognition;
+import com.amazonaws.services.rekognition.AmazonRekognitionClientBuilder;
+import com.amazonaws.services.rekognition.model.DetectLabelsRequest;
+import com.google.common.collect.ImmutableList;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.vision.core.service.VisionFeature;
+import org.nuxeo.vision.core.service.VisionProvider;
+import org.nuxeo.vision.core.service.VisionResponse;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class AmazonRekognitionProvider implements VisionProvider {
+
+    public static final String REGION_PARAM = "region";
+
+    public static final String ACCESS_KEY_PARAM = "accessKey";
+
+    public static final String SECRET_KEY_PARAM = "secretKey";
+
+    protected static final long _5MB = 5242880;
+
+    protected static final List<String> SUPPORTED_FORMAT = ImmutableList.of("image/jpeg", "image/png");
+
+    private volatile AmazonRekognition client = null;
```

Comment:
`private` -> `protected` to allow subclassing. Same in GoogleVisionProvider etc
`= null` is not needed so remove it.
Please add a comment explaining that it's `volatile` on purpose to allow for the double-checked locking idiom.